src/protocol.c

```diff
@@ -136,8 +138,16 @@ thread_run_command(void *args) {
                 perror("setenv");
                 exit(1);
             }
-            if (execvp(server->argv[0], server->argv) < 0) {
-                perror("execvp");
+            int e = -1;
+            if ( 0 == access(server->argv[0], R_OK | X_OK) ) {
```

Comment:
It s possible but the gain does not worth it at all. Are you sure you want to carry another variable in server structure to tell if the command line was quoted or not and if the first arguments need to be resolved or not ?